Fix manual task trigger failing for k8s.

snjypl opened the pull request:
[Fix](https://github.com/apache/airflow/pull/28394/commits/dfccccdd3c12ba6dd50a69f34d97cb20f1b8f200) https://github.com/apache/airflow/issues/28391 [manual task trigger from UI fails for k8s executor](https://github.com/apache/airflow/pull/28394/commits/dfccccdd3c12ba6dd50a69f34d97cb20f1b8f200) 

Manual task trigger from UI fails for k8s executor. the executor.job_id
is currently set to "manual". the task instance queued_by_job_id field
is expected to be None|Integer. this causes the filter query in
clear_not_launched_queued_tasks method in kubernetes_executor to fail
with psycopg2.errors.InvalidTextRepresentation invalid input syntax for integer: "manual" error.

setting the job_id to None fixes the issue.

Fixes: https://github.com/apache/airflow/issues/28391
<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

potiuk reviewed (changes requested):
I thikn we need a ot more context in commit - what hapens here.

snjypl committed:
Fix #28391 manual task trigger from UI fails for k8s executor

Manual task trigger from UI fails for k8s executor. the executor.job_id
is currently set to "manual". the task instance queued_by_job_id field
is expected to be None|Integer. this causes the filter query in
clear_not_launched_queued_tasks method in kubernetes_executor to fail
with psycopg2.errors.InvalidTextRepresentation invalid input syntax for integer: "manual" error.

setting the job_id to None fixes the issue.
